src/component/CodeBlock.tsx

```diff
@@ -13,18 +17,79 @@ darkTheme["hljs-selection"] = {
   backgroundColor: "#3a404b" // https://github.com/atom/one-dark-ui/blob/master/styles/ui-variables.less#L32
 };
 
+const useStyles = makeStyles(theme => ({
+  copyButton: {
+    float: 'left'
+  },
+  hidden: {
+    display: 'none'
+  },
+  copied: {
+    opacity: 0,
+    transition: 'opacity 0.3s'
+  }
+}));
+
+
 function CodeBlock(props: SyntaxHighlighterProps) {
   const darkMode = useDarkMode();
+  const classes = useStyles(darkMode);
+
+  function onContainerEnter() {
+    let newState = Object.assign({}, state)
+    newState.showCopy = true
+    setState( newState );
+  }
+
+  function onContainerLeave() {
+    let newState = Object.assign({}, state)
+    newState.showCopy = false
+    setState( newState );
+  }
+
+  function onCopy() {
+    let newState = Object.assign({}, state)
+    newState.copied = true
+    setState( newState );
+    setTimeout(() => {
+      newState = Object.assign({}, state)
+      newState.copied = false;
+      setState(newState)
+    }, 300)
+  }
+
+  const [state, setState] = React.useState({
+    showCopy: false,
+    copied: false,
+  });
```

Comment:
Could you maybe replace this with two state objects to make the state handling slightly less verbose - you can then get rid of all of the `Object.assign`s.

```typescript
const [showCopy, setShowCopy] = React.useState(false);
const [copied, setCopied] = React.useState(false)
```